pkg/core/dao/dao.go

```diff
@@ -805,18 +831,53 @@ func (dao *Simple) StoreAsTransaction(tx *transaction.Transaction, index uint32,
 	}
 	dao.Store.Put(key, buf.Bytes())
 	if dao.Version.P2PSigExtensions {
-		var value []byte
+		var (
+			valuePrefix []byte
+			newSigners  []byte
+		)
 		for _, attr := range tx.GetAttributes(transaction.ConflictsT) {
 			hash := attr.Value.(*transaction.Conflicts).Hash
 			copy(key[1:], hash.BytesBE())
-			if value == nil {
-				buf.Reset()
-				buf.WriteB(storage.ExecTransaction)
-				buf.WriteU32LE(index)
-				buf.BinWriter.WriteB(transaction.DummyVersion)
-				value = buf.Bytes()
+
+			old, err := dao.Store.Get(key)
+			if err != nil && !errors.Is(err, storage.ErrKeyNotFound) {
+				return fmt.Errorf("failed to retrieve previous conflict record for %s: %w", hash.StringLE(), err)
+			}
+			if err == nil {
+				if len(old) <= 6 { // storage.ExecTransaction + U32LE index + transaction.DummyVersion
+					return fmt.Errorf("invalid conflict record format of length %d", len(old))
+				}
+			}
+			buf.Reset()
+			buf.WriteBytes(old)
+			if len(old) == 0 {
+				if len(valuePrefix) != 0 {
+					buf.WriteBytes(valuePrefix)
+				} else {
+					buf.WriteB(storage.ExecTransaction)
+					buf.WriteU32LE(index)
+					buf.WriteB(transaction.DummyVersion)
+				}
+			}
+			newSignersOffset := buf.Len()
+			if len(newSigners) == 0 {
+				for _, s := range tx.Signers {
+					s.Account.EncodeBinary(buf.BinWriter)
+				}
+			} else {
+				buf.WriteBytes(newSigners)
+			}
+			bs := buf.Bytes()
+			val := make([]byte, len(bs))
```

Comment:
https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/util/slice#Copy ?
And you don't need it unless you're saving `valuePrefix` and `newSigners` (which you also don't need if there is just one conflict).

Reply:
> https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/util/slice#Copy

Yes, it suits here, fixed.

> And you don't need it unless you're saving valuePrefix and newSigners (which you also don't need if there is just one conflict).

Fixed for a single attribute.